src/pages/index.js

```diff
@@ -48,55 +83,130 @@ const cardListEl = new Section(
+const deleteModalConfirmation = new PopupConfirmation(
+  "#modal-confirm",
+  async (cardId, cardElement) => {
+    try {
+      await api.deleteCard(cardId);
+      cardElement.remove();
+    } catch (err) {
+      console.error(`Error On Card Deletion ${err}`);
+    }
+  }
 );
-addCardPopup.setEventListeners();
+
+function handleDeleteCard(card) {
+  deleteModalConfirmation.setSubmitAction(() => {
+    api
+      .deleteCard(card.getId())
+      .then(() => {
+        card.remove();
+        deleteModalConfirmation.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  });
+  deleteModalConfirmation.open();
+}
+
+function handleLikeClick(card) {
+  if (card._isLiked) {
+    api
+      .dislikeCard(card._id)
+      .then(() => {
+        card.toggleLike();
+        card._isLiked = false;
+      })
+      .catch((err) => {
+        console.error(`Error on Card Dislike ${err}`);
+      });
+  } else {
+    api
+      .likeCard(card._id)
+      .then(() => {
+        card.toggleLike();
+        card._isLiked = true;
+      })
+      .catch((err) => {
+        console.error(`Error on Card Like ${err}`);
+      });
+  }
+}
 
 const editProfilePopup = new PopupWithForm(
   "#profile-edit-modal",
-  handleProfileEditSubmit
+  async (data) => {
+    await api.updateUserProfile({
+      name: data.title,
+      about: data.description,
+    });
+    userInfo.setUserInfo({
+      title: data.title,
+      description: data.description,
+    });
+  }
 );
 editProfilePopup.setEventListeners();
 
-function handleProfileEditSubmit(formData) {
-  userInfo.setUserInfo({
-    name: formData.title,
-    description: formData.description,
-  });
-  editProfilePopup.close();
-}
+const addCardPopup = new PopupWithForm("#add-card-modal", async (formData) => {
+  const name = formData.title;
+  const link = formData.Url;
 
-function handleAddCardFormSubmit(inputValues) {
-  console.log(inputValues);
-  const name = inputValues.title;
-  const link = inputValues.Url;
-  const cardData = { name, link };
-  cardListEl.addItem(createCard(cardData));
-  console.log(cardData);
-  addCardPopup.close();
-  addCardFormElement.reset();
-}
+  const res = await api.createCard({ name, link });
+  cardListEl.addItems(createCard(res));
+});
+
+addCardPopup.setEventListeners();
+
+const editAvatarModal = new PopupWithForm(
+  "#edit-avatar-modal",
+  async (formData) => {
+    const avatarUrl = formData.avatar;
+    const userData = await api.updateAvatar(avatarUrl);
+    userInfo.setUserAvatar(userData.avatar);
+  }
+);
+
+editAvatarButton.addEventListener("click", () => {
+  avatarFormValidator.resetValidation();
+  editAvatarModal.open();
+});
+
+editAvatarModal.setEventListeners();
 
 profileEditButton.addEventListener("click", () => {
   const currentUserInfo = userInfo.getUserInfo();
   profileTitleInput.value = currentUserInfo.name;
   profileDescriptionInput.value = currentUserInfo.description;
+  editFormValidator.resetValidation();
```

Comment:
Good job. This is the right place for `resetValidation`. It's only for the profile form